### PR TITLE
[maintenance] eventbus

### PR DESCRIPTION
**Pull request by GrzegorzBobryk**
### First time contributor checklist
- [x] I have read [how to contribute](https://github.com/signalapp/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor License Agreement](https://whispersystems.org/cla/)

### Contributor checklist
- [x] I am following the [Code Style Guidelines](https://github.com/signalapp/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Pixel 3a, API 33
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)

----------

### Description
Eventbus renovation:
- bump,
- logging exceptions only in debug,
- eventBusIndex generated from kapt.

It will add new fixes to this lib. Also it will use generated indexes instead of using reflection. In places with private access or in anonymous classes it will work in old way. Documentation for this feature [here](https://greenrobot.org/eventbus/documentation/subscriber-index/), if you want to check it out.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -11,6 +11,7 @@ plugins {
     id 'com.squareup.wire'
     id 'translations'
     id 'licenses'
+    id 'kotlin-kapt'
```
Comment (alex-signal) [inline app/build.gradle:14]: Questioning whether we shouldn't be using KSP https://developer.android.com/build/migrate-to-ksp

I'll bring this up with the team... I'm apprehensive of adding kapt.

**Review by alex-signal — COMMENTED** [review]
I'm going to run this by the team, because bringing in an annotation processor is setting off a flag in my brain.

**Comment by alex-signal** [comment]
Pending any issues, this should go out in `6.37.x`

**Comment by GrzegorzBobryk** [comment]
> Pending any issues, this should go out in `6.37.x`

So anything to fix in this pr? I see that version `6.38.2` is released.

**Comment by greyson-signal** [comment]
Unfortunately while this build is functionally fine, it broke our reproducible builds. We're not sure why, but we're sure this change was it. Must be something non-deterministic with how kapt is used?

**Comment by GrzegorzBobryk** [comment]
Could I read some logs or anything?

**Comment by greyson-signal** [comment]
It's not like a functional thing where there's a bug with logs. The issue is that if you build two copies of the app, they are not identical. The dex's don't match. Our reproducible build instructions are [here](https://github.com/signalapp/Signal-Android/tree/main/reproducible-builds).

You can test it yourself by using our docker image (which is what we use to build release version of the app) to build multiple copies, and then you can compare them with the python script and it'll present you with the data that doesn't match. In fact, that very process is what we have run in a nightly cron job, and that's how we knew it broke :)

**Pull request closed without merging**